src/api.ts

```diff
@@ -567,7 +567,7 @@ function getModel(baseModel: Model<any>, body?: any, options?: GooseRESTOptions<
   if (!modelName) {
     return baseModel;
   } else {
-    const model = (options?.discriminatorMap ?? {})[modelName];
+    const model = (baseModel.discriminators ?? {})[modelName];
```

Comment:
Can you remove discriminatorMap from the tests and interface? Apparently not needed (woo!)